src/legacy/server/logging/index.js

```diff
@@ -6,12 +6,10 @@
  * Side Public License, v 1.
  */
 
-import { setupLogging, setupLoggingRotate, attachMetaData } from '@kbn/legacy-logging';
+import { setupLogging, setupLoggingRotate } from '@kbn/legacy-logging';
 
 export async function loggingMixin(kbnServer, server, config) {
-  server.decorate('server', 'logWithMetadata', (tags, message, metadata = {}) => {
-    server.log(tags, attachMetaData(message, metadata));
-  });
+  server.decorate('server', 'logWithMetadata');
 
```

Comment:
[Removing the log](https://github.com/elastic/kibana/issues/82431#issuecomment-794211967) was also part of the tag-on request for deprecating legacy logging configuration.

Reply:
Ah, sorry, I had missed that comment from the issue. Thanks for clarifying.

In that case I think my note (2) still applies: without a 3rd argument, `decorate` is just going to set `logWithMetadata` to `undefined`.

It feels like we can probably remove that call entirely, as well as the `kbn_server.d.ts` reference I mentioned, but I'd defer to what @restrry thinks on this as he's more familiar with it.

(Alternatively we could add a no-op function if BWC is at all a concern here, so that you could still call `logWithMetadata(...)` and it would just do nothing)

Reply:
@lukeelmers would this work?:
```
// eslint-disable-next-line no-unused-vars
server.decorate('server', 'logWithMetadata', (tags, message, metadata = {}) => {});
```

Reply:
@TinaHeiligers Yeah, I guess what I'm proposing is one of two things:
- If we are okay removing `logWithMetadata` entirely, just remove it.
- Or if we want to keep it, make it a no-op as in your example.

The reason being that if we kept this without a 3rd parameter, `logWithMetadata` becomes `undefined` and so if there _were_ any 3rd party plugins relying on this, calling `logWithMetadata()` would suddenly throw a type error. At which point we might as well remove it entirely as it would already be a breaking change.

Reply:
I removed `logWithMetadata` entirely (also removed from `kbn_server.d.ts`)